src/NuGet.Core/NuGet.DependencyResolver/SourceRepositoryDependencyProvider.cs

```diff
@@ -139,9 +140,21 @@ private IEnumerable<LibraryDependency> GetDependencies(FindPackageByIdDependency
         {
             if (_findPackagesByIdResource == null)
             {
-                _findPackagesByIdResource = await _sourceRepository.GetResourceAsync<FindPackageByIdResource>();
-                _findPackagesByIdResource.Logger = _logger;
-                _findPackagesByIdResource.CacheContext = _cacheContext;
+                // Avoid multiple instances of the same resource
+                await _resourceLock.WaitAsync();
```

Comment:
Async was the reason. Now that there is only one httpsource I don't think thread safety is important here. It could be reverted back to how it was.